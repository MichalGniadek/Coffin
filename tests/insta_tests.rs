use coffin2::{
    ast::Ast,
    debug_print::DebugPrint,
    lexer::Token,
    name_resolution::NameResolution,
    parser::{self, spans_table::SpansTable},
};
use insta::{assert_snapshot, glob};
use lasso::RodeoResolver;
use logos::Logos;
use std::{fs, path::Path};

fn get_ast(path: &Path) -> (Ast, SpansTable, RodeoResolver) {
    let code = fs::read_to_string(path).unwrap();
    let lexer = Token::lexer(&code);
    parser::parse(lexer)
}

#[test]
fn insta() {
    glob!(r"insta_parser\*.coff", |path| {
        let (ast, _, rodeo) = get_ast(path);
        assert_snapshot!(DebugPrint::visit(&ast, Some(&rodeo), None, None));
    });
    glob!(r"insta_parser_spans\*.coff", |path| {
        let (ast, spans, rodeo) = get_ast(path);
        assert_snapshot!(DebugPrint::visit(&ast, Some(&rodeo), Some(&spans), None));
    });
    glob!(r"insta_variables\*.coff", |path| {
        let (ast, _, rodeo) = get_ast(path);
        let vars = NameResolution::visit(&ast);
        assert_snapshot!(DebugPrint::visit(&ast, Some(&rodeo), None, Some(&vars)));
    });
}
