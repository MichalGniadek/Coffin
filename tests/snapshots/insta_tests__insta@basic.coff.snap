---
source: tests/insta_tests.rs
expression: "DebugPrint::visit(&ast, Some(&rodeo), None, Some(&vars))"

---
fun fn0@X (arg0@0 : T@X, arg1@1 : T@X)  {
	(let var0@2 = #0)
	(var0@2 = $arg0@0)
	{
		(arg0@0 = #1)
		(let arg0@3 = #2)
		(arg0@3 = #3)
	}
	(arg0@0 = #4)
	(let var1@4 = #5)
	(var1@4 = #6)
	(let var1@5 = #7)
	(var1@5 = #8)
	(arg0@0 = #9)
}
fun fn1@X (arg2@6 : T@X)  {
	(let var1@7 = #0)
	(var1@7 = ($var1@7 + #1))
	(let var1@8 = ($var1@7 + #1))
	(var1@8 = ($var1@8 + #1))
}
